sonaranalyzer-dotnet/tests/SonarAnalyzer.UnitTest/Helpers/SonarAnalysisContextTest.cs

```diff
@@ -64,6 +64,32 @@ public void SonarAnalysis_NoIssueReportedIfAnalysisIsDisabled()
             }
         }
 
+        [TestMethod]
+        public void SonarAnalysis_IsAbleToDisableSpecificRules()
+        {
+            SonarAnalysisContext.ShouldAnalysisBeDisabled = (tree, descriptor) =>
+                descriptor.Id == AnonymousDelegateEventUnsubscribe.DiagnosticId;
+
+            try
+            {
+                foreach (var testCase in TestCases)
+                {
+                    if (testCase.Analyzer.GetType() == typeof(AnonymousDelegateEventUnsubscribe))
+                    {
+                        Verifier.VerifyNoIssueReported(testCase.Path, testCase.Analyzer);
+                    }
+                    else
+                    {
+                        Verifier.VerifyAnalyzer(testCase.Path, testCase.Analyzer);
+                    }
+                }
+            }
+            finally
+            {
+                SonarAnalysisContext.ShouldAnalysisBeDisabled = null;
```

Comment:
Consider using setup/teardown methods instead, seems cleaner to me.

Reply:
Looks like something for a next commit.